File: backend/src/main/java/com/bayerna/reserveforces/domain/user/UserAccount.java
package com.bayerna.reserveforces.domain.user;

import com.bayerna.reserveforces.domain.reservist.Reservist;
import com.bayerna.reserveforces.domain.unit.Unit;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 사용자계정 (USER_ACCOUNT). docs/db-design.md 참조.
 * 관리자 계정은 unit에 소속되고, 일반(예비군) 계정은 reservist와 1:1로 연결된다.
 */
@Entity
@Table(name = "user_account")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "user_id")
    private Long userId;

    @Column(name = "login_id", nullable = false, unique = true, length = 50)
    private String loginId;

    @Column(name = "password_hash", nullable = false, length = 255)
    private String passwordHash;

    @Enumerated(EnumType.STRING)
    @Column(name = "role", nullable = false, length = 20)
    private Role role;

    @ManyToOne
    @JoinColumn(name = "unit_id")
    private Unit unit;

    @ManyToOne
    @JoinColumn(name = "reservist_id")
    private Reservist reservist;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @jakarta.persistence.PrePersist
    void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: backend/src/main/kotlin/com/bayerna/reserveforces/domain/user/UserAccount.kt
package com.bayerna.reserveforces.domain.user

import com.bayerna.reserveforces.common.ActiveStatus
import com.bayerna.reserveforces.common.TimestampedEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Table
import java.time.OffsetDateTime
import java.util.UUID

/** users (사용자계정). docs/db-design.md 5.1 참조. */
@Entity
@Table(name = "users")
class UserAccount(
    @Column(name = "login_id", nullable = false, unique = true, length = 50)
    var loginId: String,

    @Column(name = "password_hash", nullable = false, length = 255)
    var passwordHash: String,

    @Column(name = "name", nullable = false, length = 50)
    var name: String,

    @Enumerated(EnumType.STRING)
    @Column(name = "role", nullable = false, length = 20)
    var role: UserRole,

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    var status: ActiveStatus = ActiveStatus.ACTIVE,

    @Column(name = "last_login_at")
    var lastLoginAt: OffsetDateTime? = null,

    id: UUID = UUID.randomUUID(),
) : TimestampedEntity(id)
